### PR TITLE
fix: Generate the return url from universallink helper

### PR DESCRIPTION
**Pull request by Crash--**
Repeat after me: don't use window.location.href to get access to the current url. It doesn't work on mobile 🤦‍♂ (file:///....../index.html). 

Instead, use the UI helper to generate the correct Universal Link 

Toolbar Menu: we've done some rework of the Toolbar item to create some "features groups". First one : creation (new note, new folder, upload files, scan file). Second : Folder's action : download / share / select items. Third: Deletion

### CODE REVIEW

File: src/drive/web/modules/drive/Toolbar.jsx [hunk 1]
```diff
@@ -87,6 +82,11 @@ class Toolbar extends Component {
             </Item>
           )}
         <hr />
+        <NotRootFolder>
+          <Item>
+            <ShareItem />
+          </Item>
+        </NotRootFolder>
```
Comment (ptbrowne) [inline src/drive/web/modules/drive/Toolbar.jsx]: Why this reordering ? Is it intended ?

File: src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
 import styles from 'drive/styles/toolbar.styl'
-import { withClient, models, useAppLinkWithStoreFallback } from 'cozy-client'
+import useCapabilities from 'lib/hooks/useCapabilities'
 
 const CreateNoteItem = ({ client, t, displayedFolder }) => {
   const { fetchStatus, url, isInstalled } = useAppLinkWithStoreFallback(
     'notes',
     client
   )
+  const capabilities = useCapabilities(client)
```
Comment (ptbrowne) [inline src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: Friendly reminder that useCapabilities does not use a fetchPolicy meaning that we fetch the settings every time a CreateNoteItem is displayed.

File: src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx [hunk 3]
```diff
@@ -1,16 +1,31 @@
...
+    capabilities,
+    'capabilities.attributes.flat_subdomains'
+  )
 
+  const returnUrl = generateUniversalLink({
+    slug: 'drive',
+    cozyUrl: client.getStackClient().uri,
+    subDomainType: isFlatDomain ? 'flat' : 'nested',
+    nativePath: `/files/${displayedFolder ? displayedFolder.id : null}`
+  })
```
Comment (ptbrowne) [inline src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: Could you add a test for this 😗 ?

File: src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx (hunk 2 above)
Comment (Crash--) [reply src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: It's displayed all the time, so only one request or one more if the client change ;). 

I'm wondering if I should move this hook to cozy-client in order to not repeat that everywhere and also if one day we have a useQuery() with fetch policy we could be able to update all our app with that :)

File: src/drive/web/modules/drive/Toolbar.jsx (hunk 1 above)
Comment (Crash--) [reply src/drive/web/modules/drive/Toolbar.jsx]: Yep. PR's description updated

File: src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx (hunk 2 above)
Comment (ptbrowne) [reply src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: I think it has its place in cozy-client, yes :)

File: src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx [hunk 4]
```diff
@@ -25,7 +40,7 @@ const CreateNoteItem = ({ client, t, displayedFolder }) => {
           window.location.href = await models.note.generatePrivateUrl(
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: Is it ok to use `window.location.href` for the redirection? Shouldn't we use `open(url)` or something similar?

Comment (Crash--) [reply src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: Yep, it's ok. Our apps are not configured to handle this kind of opening, so it opens the browser.

File: src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx (hunk 3 above)
Comment (Crash--) [reply src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: I'm not convinced by this test, I just write a defensive code, and I don't know if it worth it

Comment (Crash--) [reply src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx]: If you really want this test, I'll try to take time to write few others for the toolbar.

**Pull request merged**